pydantic/_internal/_repr.py

```diff
@@ -101,6 +101,13 @@ def display_as_type(obj: Any) -> str:
     elif isinstance(obj, Representation):
         return repr(obj)
 
+    if isinstance(obj, str):
+        # TODO: I'm hitting this for generic forward references. It used to return 'str' in this case. Affects:
+        #   * tests/test_json_schema.py::test_nested_generic
+        #   * tests/test_utils.py::test_display_as_type[foobar-str]
+        #   * tests/test_utils.py::test_display_as_type_310[<lambda>-str1]
+        return obj
```

Comment:
@samuelcolvin Also interested to get your eyes on this

Reply:
I'm optimistically removing this comment from the PR to get CI run through with the comment removed, in the hopes that you agree with the change. To make it easier to review, the comment that used to be above this line said:
```
        # TODO: I'm hitting this for generic forward references. It used to return 'str' in this case. Affects:
        #   * tests/test_json_schema.py::test_nested_generic
        #   * tests/test_utils.py::test_display_as_type[foobar-str]
        #   * tests/test_utils.py::test_display_as_type_310[<lambda>-str1]
```

Reply:
I'll also note that if for any reason we _don't_ want to make this change to this function, I can change the `model_parametrized_name` method to have `param_names = [param if isinstance(param, str) else _repr.display_as_type(param) for param in params]`

Reply:
happy to return here, but I wonder if we should return `f'ForwardRef({obj!r})'`?

Advtanges:
* it's hopefully clearer what we think is going on
* It would better match the behaviour of `List['SomeForwardRef']` which becomes `"List[ForwardRef('SomeForwardRef')]"` as per the test in `test_utils.py`